### PR TITLE
buf-language-server: init at unstable-2022-08-19

### PR DESCRIPTION
**Pull request by svrana**
###### Description of changes

bufls is a prototype for the beginnings of a Protobuf language server compatible with Buf modules and workspaces
https://github.com/bufbuild/buf-language-server

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by svrana** [comment]
Result of `nixpkgs-review pr 189307` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>buf-language-server</li>
  </ul>
</details>

File: pkgs/development/tools/buf-language-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    rev = version;
+    sha256 = "sha256-UHsWrWDOC/f3YS2g533CgUkuUmz4MUQRunClQiY/YPQ=";
+  };
+
+  vendorSha256 = "sha256-ORzCOmBx6k1GZj6pYLhqPsdneCc7Tt1yHpI5mw5ruFU=";
+
+  meta = with pkgs.lib; {
+    description = "Language server for protocol buffers";
+    license = licenses.asl20;
+    platforms = platforms.unix;
```
Comment (azahi) [inline pkgs/development/tools/buf-language-server/default.nix:19]: This is the default.
```suggestion
```

File: pkgs/development/tools/buf-language-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchFromGitHub {
+    owner = "bufbuild";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-UHsWrWDOC/f3YS2g533CgUkuUmz4MUQRunClQiY/YPQ=";
+  };
+
+  vendorSha256 = "sha256-ORzCOmBx6k1GZj6pYLhqPsdneCc7Tt1yHpI5mw5ruFU=";
+
+  meta = with pkgs.lib; {
```
Comment (azahi) [inline pkgs/development/tools/buf-language-server/default.nix:16]: Missing `homepage`.

File: pkgs/development/tools/buf-language-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ pkgs, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
```
Comment (azahi) [inline pkgs/development/tools/buf-language-server/default.nix:3]: Missing [ldflags](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#ldflags-var-go-ldflags).

File: pkgs/development/tools/buf-language-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "buf-language-server";
+  version = "6f08a7eed22c5a178cb55613f454319e09be112c";
```
Comment (azahi) [inline pkgs/development/tools/buf-language-server/default.nix:5]: This should be `unstable-YYYY-MM-DD` if no upstream tag is specified.

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 189307` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>buf-language-server</li>
  </ul>
</details>

File: pkgs/development/tools/buf-language-server/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{ pkgs, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "buf-language-server";
+  version = "unstable-2022-09-01";
+
+  src = fetchFromGitHub {
+    owner = "bufbuild";
+    repo = pname;
+    rev = version;
```
Comment (azahi) [inline pkgs/development/tools/buf-language-server/default.nix:10]: You need to specify a commit ref here.

Comment (svrana) [reply pkgs/development/tools/buf-language-server/default.nix]: Ha, got it! Thanks @azahi !

File: pkgs/development/tools/buf-language-server/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ pkgs, buildGoModule, fetchFromGitHub }:
```
Comment (zowoq) [inline pkgs/development/tools/buf-language-server/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/development/tools/buf-language-server/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  vendorSha256 = "sha256-ORzCOmBx6k1GZj6pYLhqPsdneCc7Tt1yHpI5mw5ruFU=";
+
+  ldflags = [
+    "-s"
+    "-w"
+  ];
+
+  meta = with pkgs.lib; {
```
Comment (zowoq) [inline pkgs/development/tools/buf-language-server/default.nix:21]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/tools/buf-language-server/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "buf-language-server";
+  version = "unstable-2022-09-01";
```
Comment (zowoq) [inline pkgs/development/tools/buf-language-server/default.nix:5]: Sorry, didn't notice this at first.

```suggestion
  version = "unstable-2022-08-19";
```


> https://github.com/NixOS/nixpkgs/blob/master/doc/contributing/coding-conventions.chapter.md#package-naming-sec-package-naming
>
> If a package is not a release but a commit from a repository, then the `version` attribute _must_ be the date of that (fetched) commit. The date _must_ be in `"unstable-YYYY-MM-DD"` format.

Comment (svrana) [reply pkgs/development/tools/buf-language-server/default.nix]: ahh, thanks for this link.

**Pull request merged**